library/core/src/ptr/mod.rs

```diff
@@ -56,6 +56,48 @@
 //! has size 0, i.e., even if memory is not actually touched. Consider using
 //! [`NonNull::dangling`] in such cases.
 //!
+//! ## Pointer to reference conversion
+//! When converting a pointer to a reference `&T` using `&*`,
+//! there are several rules that must be followed:
+//!
+//! * The pointer must be properly aligned.
+//!
+// some microprocessors may use address 0 for an interrupt vector.
+// users of these microprocessors must always read/write address 0 through
+// a raw pointer, not a reference.
+//! * It must be non-null.
+//!
+//! * It must be "dereferenceable" in the sense defined above.
+//!
+//! * The pointer must point to a valid instance of `T`.
+//!   This means that the created reference can only refer to
+//!   uninitialized memory through careful use of `MaybeUninit`.
```

Comment:
This is not quite true. Uninit memory in padding is also okay, and inside (some) `union`s it is okay, too.

This should probably just link to the reference for the definition of "valid". (Also, I don't think we use the term `instance` usually. The reference says "valid value of type `T`".)

Reply:
the rust reference does not seem to have a general definition for "valid", it only has sections on bit validity for certain types.

Reply:
It does, [here](https://doc.rust-lang.org/reference/behavior-considered-undefined.html). With https://github.com/rust-lang/reference/pull/1540 that becomes a separate section that can even be hyperlinked.

Reply:
that pr got merged 2 days ago, but the new section doesn't even show up in the nightly reference.

any idea what the staging cycle is for the reference? if it's relativly short i'm fine blocking this PR on that getting updated so it can include that link, which is quite useful.

Reply:
Not sure, @ehuss is doing book updates every now and then. @ehuss how often do you usually do that?

Reply:
Every 2 weeks.

I'm about to open a PR to update it, but it might take a while to get it reviewed.